pkg/identityserver/entity_access.go

```diff
@@ -245,6 +246,25 @@ func (is *IdentityServer) RequireAuthenticated(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
+	if userID := authInfo.GetEntityIdentifiers().GetUserIDs(); userID != nil {
+		err = is.withDatabase(ctx, func(db *gorm.DB) (err error) {
+			user, err := store.GetUserStore(db).GetUser(ctx, userID, &types.FieldMask{Paths: []string{
+				"state",
+			}})
+			if err != nil {
+				return err
+			}
+			if user.State != ttnpb.STATE_APPROVED {
+				return errUnnaprovedUser
```

Comment:
I don't think we should use the term "unapproved" for all cases. It would be better to do a switch similar to the one in https://github.com/TheThingsNetwork/lorawan-stack/blob/4af802d1fb06e54e21de16fc7f34b6c829f5afe8/pkg/identityserver/entity_access.go#L216-L235

Reply:
Does this mean that users with `STATE_FLAGGED` should also pass the authentication check similar to `STATE_APPROVED` ?